### PR TITLE
[NVIDIA]Fix local_num_experts for EP

### PR DESCRIPTION
**Pull request by wenscarl**
## Motivation

This PR fixes a bug in https://github.com/sgl-project/sglang/pull/8552. At create_weight, the num_experts and num_loca_experts should both be passed in for EP case.

@kaixih @kushanam 

cc. @kushanam
## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Comment by zhyncs** [comment]
Hi @wenscarl this pr breaks unit-test-deepep-4 so I reverted https://github.com/sgl-project/sglang/pull/8797

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 1]
```diff
@@ -752,6 +752,7 @@ def create_weights(
         self,
         layer: torch.nn.Module,
         num_experts: int,
```
Comment (kaixih) [inline python/sglang/srt/layers/quantization/modelopt_quant.py:754]: Why not simply change the num_experts to num_local_experts? I think we just need the actual num of experts when allocating weights, right?
